linera-views/src/memory.rs

```diff
@@ -16,69 +17,37 @@ use thiserror::Error;
 pub type MemoryStoreMap = BTreeMap<Vec<u8>, Vec<u8>>;
 
 /// A virtual DB client where data are persisted in memory.
-pub type MemoryClient = Arc<RwLock<MutexGuardArc<MemoryStoreMap>>>;
-
-/// An implementation of [`crate::common::Context`] that stores all values in memory.
-pub type MemoryContext<E> = ContextFromDb<E, MemoryClient>;
-
-impl<E> MemoryContext<E> {
-    /// Creates a [`MemoryContext`].
-    pub fn new(guard: MutexGuardArc<MemoryStoreMap>, extra: E) -> Self {
-        Self {
-            db: Arc::new(RwLock::new(guard)),
-            base_key: Vec::new(),
-            extra,
-        }
-    }
-}
-
-/// Provides a `MemoryContext<()>` that can be used for tests.
-pub fn create_test_context() -> MemoryContext<()> {
-    let state = Arc::new(Mutex::new(BTreeMap::new()));
-    let guard = state
-        .try_lock_arc()
-        .expect("We should acquire the lock just after creating the object");
-    MemoryContext::new(guard, ())
-}
-
-/// Creates a test memory client.
-pub fn create_test_memory_client() -> MemoryClient {
-    let state = Arc::new(Mutex::new(BTreeMap::new()));
-    let guard = state
-        .try_lock_arc()
-        .expect("We should acquire the lock just after creating the object");
-    Arc::new(RwLock::new(guard))
+#[derive(Clone)]
+pub struct BasicMemoryClient {
```

Comment:
Why not define `MemoryClient` directly like this?

Reply:
The reason why `BasicMemoryClient` is not a `KeyValueStoreClient` is that the function `write_batch` takes an additional argument for controlling the size of the values and throwing an error if that happens.

By doing this in this this way, I achieve the following:
* I limit the code duplication in the construction of `MemoryClient` and `TestMemoryClient`: Both rely on `BasicMemoryClient` for their operation.
* I enforce the limit on the size as I should have done.

Reply:
see comment below

Reply:
Ok, I eliminated the `BasicMemoryClient` and simplified the code.